### ISSUE TITLE
Pulse report was received at wrong time

### ISSUE THREAD

**Issue by ansidev** [metabase/metabase#6698]
#### Environment Information
- Chrome Version 63.0.3239.132 (Official Build) (64-bit)
- macOS 10.13.2
- MariaDB
- Metabase 0.27.2
- Hosting: Docker on Google Cloud Platform
- Metabase internal database: MariaDB
#### Step to reproduce the bug
- Create a daily Slack pulse at 6:00 AM => Slack messages will be sent at 6:00 AM ICT, your Metabase timezone.
- But I received pulse at 1:00 PM ICT on Slack.
-------

**Comment by senior** [comment]
I think I need some more information, because the code looks to be doing the right thing. Can you confirm that you have set your Metabase report timezone (in the General section of the admin panel) to ICT? It looks like it's sending the emails in UTC time, which might be the timezone your server (or JVM instance) is in. If a report timezone isn't specified, Metabase will use the JVM timezone. The JVM timezone  is usually inherited from the system it's running on unless you've provided the `user.timezone` system property when you launch Metabase.

**Comment by ansidev** [comment]
I confirm that there are no problems with timezone, but it takes some hours to sync timezone. And there is no option for Vietnam when I choose timezone in General section of the admin panel.

**Comment by ansidev** [comment]
I provide more detail info.

I run Metabase in a Docker container.

### Timezone configurations:

- Database server: **Asia/Ho_Chi_Minh** (aka UTC+7)
- Metabase server: docker environment variable JAVA_TIMEZONE = **Asia/Ho_Chi_Minh**
- Container contains Metabase server: **Asia/Ho_Chi_Minh**
- Report timezone in Admin panel: **Don't set** because there is no option named Asia/Ho_Chi_Minh.

### Problem

- Display time for query result deviation: **+7 hours** ~ **real time +14 hours**.
- Send report time deviation: **+7 hours** ~ **real time +14 hours**.

Hope this bug was solved ASAP!

**Comment by senior** [comment]
@ansidev Ok, I think I see your configuration issue. Instead of `JAVA_TIMEZONE`, you should be setting the Java system property `user.timezone`. If you are using the Metabase docker container, the easiest way to do this is to set the `JAVA_OPTS` environment variable to "-Duser.timezone". Since `JVM_TIMEZONE` has no affect and you haven't set the report timezone, the system timezone is being used. Google Cloud docs say that [UTC is the default timezone](https://cloud.google.com/container-optimized-os/docs/resources/faq) for their instances which explains why your reports were 7 hours off.

**Comment by ansidev** [comment]
Today, my team detected 1 critical bugs:

### Test environment:
- Metabase version: **0.28.1**.
- Saved time in database has timezone (Asia/Ho_Chi_Minh) UTC+7

- Run 3 docker containers:
```
docker run -d \
--name metabase-1 \
-p 3001:3000 \
metabase/metabase
```
```
docker run -d \
--name metabase-2 \
-p 3002:3000 \
-e "JAVA_TIMEZONE=Asia/Ho_Chi_Minh" \
metabase/metabase
```
```
docker run -d \
--name metabase-3 \
-p 3003:3000 \
-e JAVA_OPTS="-Duser.timezone=Asia/Ho_Chi_Minh" \
metabase/metabase
```

### Bugs

Test query: `SELECT NOW()`.

Test cases:

Go to Metabase Admin Panel.
Default report timezone setting from Metabase before I change.
1. Change report timezone to Asia/Hong_Kong (UTC+8).
2. Change report timezone to Asia/Bangkok (UTC+7).
3. Change report timezone to Database Default.

It's OK when switching between 2 timezone in Admin panel, but if you switch to a specific timezone (ex. Asia/Bangkok), then switch to Database default timezone. Metabase will use hour deviation of the previous saved report timezone compare to UTC timezone and plus database timezone instead of database timezone.

In my test case, previous saved report timezone is Asia/Bangkok (UTC+7) so deviation is _+7_ hour
Container `metabase-1`:
Timezone = UTC, deviation = **+0** => Display time = **+0** + _+7_ = +7 hours
Expected timezone: UTC => wrong display time.

Container `metabase-2`:
Timezone = UTC+7, deviation = **+7** => Display time = **+7** + _+7_ = +14 hours
Expected timezone: UTC+7 => wrong display time.

Container `metabase-3`:
Timezone = UTC+7, deviation = **+7** => Display time = **+7** + _+7_ = +14 hours
Expected timezone: UTC+7 => wrong display time.

**Comment by orware** [comment]
I think something I've been noticing with the Pulses I've created recently (v0.30.1) falls under this issue as well, but our setup is a little different/more basic (we're running the JAR on Windows).

When creating a Pulse it says the following:
"Emails will be sent at 8:00 AM PDT, your Metabase Timezone."

Which sounds great to me since we're in the Pacific Timezone and that's the time I'd like the Pulse to show up. However, when the Pulse actually runs it's being sent to me around 1 AM so it kind of seems like it's sending the Pulse out using the UTC timezone since at 1 AM Pacific Time that translates to 8 AM UTC.

I didn't have any specific timezone set in the Metabase General Configuration so it would have been defaulting to the default system timezone according to what it describes there (but the server is set to Pacific Time as well already). I've manually changed the timezone here to US/Pacific and I'll see if that changes the behavior at all.

**Comment by orware** [comment]
Just a quick update on my test.

So after changing the Metabase timezone in the General Settings area to US/Pacific, today I received another Pulse at the correct/expected time of 8 AM Pacific (rather than the previously reported 1 AM time).

Still a bit confusing overall though since the Pulse itself was showing me the correct timezone when I created it, so it wasn't immediately obvious that the issue was due to the general Metabase timezone setting to me.
